### PR TITLE
Reorganise classifier

### PR DESCRIPTION
**Pull request by eatyourgreens**
Describe your changes.
Splits Classifier and ClassifierWrapper into seperate files.

Rewrites Classifier in ES6 and adds prop type validation.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? https://reorganise-classifier.pfe-preview.zooniverse.org/
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

File: app/classifier/classifier.jsx [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+  toggleExpertClassification(value) {
+    this.setState({ showingExpertClassification: value });
+  }
+
+  changeDemoMode(demoMode) {
+    this.props.onChangeDemoMode(demoMode);
+  }
+
+  subjectIsGravitySpyGoldStandard() {
+    this.props.workflow.configuration.gravity_spy_gold_standard && this.props.subject.metadata['#Type'] === 'Gold';
```
Comment (srallen) [inline app/classifier/classifier.jsx]: This needs a `return` for the gravity spy feedback to work.

**Comment by srallen** [comment]
Just as a side note, I don't think this branch is breaking it, but I'm still not totally 100% clear on that since it seems ok on the current production site. The expert classifications with the Picker Practice workflow in Microscopy Masters seems to be buggy still on the staged branch. I'm seeing a `this.setState is not a function` being thrown when trying to toggle the expert classification in the summary. There is an expert classification set in the state, so it should work. Not sure what's going on yet.

**Comment by eatyourgreens** [comment]
Thanks. I'd forgotten to bind the `toggleExpertClassification` event handler. That should be fixed now, along with the other comments.

File: app/classifier/classifier.jsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ export default class Classifier extends React.Component {
     this.handleAnnotationChange = this.handleAnnotationChange.bind(this);
     this.handleSubjectImageLoad = this.handleSubjectImageLoad.bind(this);
     this.completeClassification = this.completeClassification.bind(this);
+    this.toggleExpertClassification = this.toggleExpertClassification(this);
```
Comment (srallen) [inline app/classifier/classifier.jsx]: Typo here, should be `this.toggleExpertClassification.bind(this)`

Comment (eatyourgreens) [inline app/classifier/classifier.jsx]: D'oh! Fixed.

**Pull request merged**